File: backend/src/branding.ts
/** Cấu hình thương hiệu — lưu trong app_settings, sửa ở Admin, tiêm vào HTML. */
import { getSetting } from './db.js';

export const BRAND_DEFAULTS: Record<string, string> = {
  brand_name: 'ANS Music',
  brand_short: 'ANS',
  brand_suffix: 'Music',
  brand_tagline: 'Nghe nhạc trực tuyến',
  brand_accent: '#7c5cff',
  brand_logo_url: '/static/img/logo.svg',
  brand_favicon_url: '/favicon.svg',
};
export const BRAND_EDITABLE = new Set(Object.keys(BRAND_DEFAULTS));

export function getBrand(): Record<string, string> {
  const out: Record<string, string> = {};
  for (const [k, def] of Object.entries(BRAND_DEFAULTS)) out[k] = getSetting(k, def) as string;
  return out;
}

File: legacy-python/app/branding.py
"""Cấu hình thương hiệu (brand) — lưu trong app_settings, sửa được ở Admin.

Được tiêm thẳng vào HTML (window.ANS_CFG.brand) nên áp dụng tức thì lúc tải
trang, không tốn request phụ và không nháy giao diện mặc định.
"""
import sqlite3

from .db import get_setting

BRAND_DEFAULTS = {
    "brand_name": "ANS Music",
    "brand_short": "ANS",              # phần trước, tô đậm màu accent
    "brand_suffix": "Music",           # phần sau
    "brand_tagline": "Nghe nhạc trực tuyến",
    "brand_accent": "#7c5cff",
    "brand_logo_url": "/static/img/logo.svg",
}

# Chỉ các key này được phép ghi qua Admin (whitelist)
EDITABLE = set(BRAND_DEFAULTS.keys())


def get_brand(conn: sqlite3.Connection) -> dict:
    return {k: get_setting(conn, k, default) for k, default in BRAND_DEFAULTS.items()}
